src/modules/display.c

```diff
@@ -34,7 +35,13 @@ static void receive(const msg_t *const msg, UNUSED const void* userdata) {
                 state.display_state |= DISPLAY_DIMMED;
                 DEBUG("Entering dimmed state...\n");
                 old_pct = state.current_bl_pct;
-                dim_backlight(conf.dim_conf.dimmed_pct);
+                /* Don't touch backlight if a lower level is already set */
```

Comment:
I'd avoid dim_skipped variable at all: just avoid storing old_pct if current backlight is below dimmed_pct level; this way we save both dim_skipped variable and restore code remains same as before.

Reply:
I believe I've reworked it to function the way you want in the latest commit, lmk what you think.